daemon/daemon.go

```diff
@@ -27,6 +28,12 @@ func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
 
 	r := mux.NewRouter().StrictSlash(true)
 
+	e, err := engine.New(cfg)
```

Comment:
defer a Close to it to clean up after there's a signal to stop?

Reply:
Added `engine` to `daemon`, so now it is handled in `srv.Shutdown`